gprofiler/spark/sampler.py

```diff
@@ -93,16 +93,18 @@ def _get_yarn_config_property(
                         return value_property.text
         return default
 
-    def _guess_driver_application_master_address(self, process: psutil.Process) -> str:
+    def _guess_standalone_master_webapp_address(self, process: psutil.Process) -> str:
         """
-        Selects the master address for a org.apache.spark.deploy.master.Master running on this node.
-        Uses master_address if given, or defaults to my hostname.
+        Selects the master address for a standalone cluster.
+        Uses master_address if given.
         """
-        if self._master_address is not None:
+        if self._master_address:
             return self._master_address
         else:
-            host_name = get_hostname()
-            return host_name + ":4040"
+            master_process_args = process.cmdline()
+            master_ip = master_process_args[master_process_args.index("--host") + 1]
+            master_port = master_process_args[master_process_args.index("--webui-port") + 1]
```

Comment:
Can the values for these args also be specified in the equals `=` form? e.g. `--host=10.1.2.3`, or `--webui-port=12345`?

Reply:
I've never witnessed such case, need to dig a bit to figure out
nice point

Reply:
seems like it does not